src/components/app/routes/loaders/rootLoader.ts

```diff
@@ -33,33 +35,56 @@ const makeRootLoader: Types.MakeRouteLoaderFunctionWithQueryClient = function ma
 
     // Retrieve linked enterprise customers for the current user from query cache
     // or fetch from the server if not available.
-    const enterpriseLearnerData = await queryClient.ensureQueryData<Types.EnterpriseLearnerData>(
-      queryEnterpriseLearner(username, enterpriseSlug),
+    const matchedBFFQuery = resolveBFFQuery(
+      requestUrl.pathname,
     );
+    let enterpriseLearnerData: Types.EnterpriseLearnerData | any;
+    if (matchedBFFQuery) {
+      enterpriseLearnerData = await queryClient.ensureQueryData(
+        matchedBFFQuery({ enterpriseSlug }),
+      );
+    } else {
+      enterpriseLearnerData = await queryClient.ensureQueryData(
+        queryEnterpriseLearner(username, enterpriseSlug),
+      );
+    }
+
     let {
       enterpriseCustomer,
       activeEnterpriseCustomer,
       allLinkedEnterpriseCustomerUsers,
     } = enterpriseLearnerData;
-    const { staffEnterpriseCustomer, enterpriseFeatures } = enterpriseLearnerData;
+    const {
+      staffEnterpriseCustomer, shouldUpdateActiveEnterpriseCustomerUser,
+    } = enterpriseLearnerData;
 
     // User has no active, linked enterprise customer and no staff-only customer metadata exists; return early.
     if (!enterpriseCustomer) {
       return null;
     }
 
+    let updateActiveEnterpriseCustomerUserResult: {
+      enterpriseCustomer: any; updatedLinkedEnterpriseCustomerUsers: any;
+    } | null = null;
```

Comment:
Bit of a code smell to use `any`. Should these types by `Types.EnterpriseCustomer` and `Types.EnterpriseCustomerUser[]`, respectively?

Reply:
Moot in refactor